serverless/serverless-azure-functions
# fix: use webpack output files on offline command

## What did you implement:
Use webpack output files (if exists) on offline command

Closes https://github.com/serverless/serverless-azure-functions/issues/392 https://github.com/serverless/serverless-azure-functions/issues/366



## How did you implement it:
`serverless-webpack` have a trigger on `before:offline:start`, So I changed offline hook so we will have `offline:start` hook.
Also, change the generated `function.json` `scriptFile` property to point to `.webpack` folder
Everything should work the same as before, just when running offline command webpack plugin (if exists) will compile the files 



## How can we verify it:

```
cd examples/typescript-webpack
npm install
npm run offline
```

You should see the error:
![image](https://user-images.githubusercontent.com/6559921/83361325-c5f4e300-a390-11ea-8408-118f1400f476.png)
This is the current behavior.

To fix it we need to link the new version of the library.
go to the root directory and run `npm link`
then go back to the example folder and run `npm link serverless-azure-functions`

`npm run offline`

![image](https://user-images.githubusercontent.com/6559921/83361327-d1480e80-a390-11ea-85fe-d5aaf258b112.png)

**AFTER releasing a new version with this fix we should update the version in the example folder so it will work without the need to run npm link**



## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
       **Validate via `npm test`**

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO
